### ISSUE TITLE
Possible to sync in two steps instead of one step?

### ISSUE THREAD

**Issue by simonhf** [imapsync/imapsync#183]
The example command syncs IMAP server A to IMAP server B. Is it possible (and how?) can I split this into 2 steps; 1. Sync all emails from IMAP server A to local disk. 2. Sync emails from server A on local disk to IMAP server B?

Also, assuming steps 1 and 2 are possible somehow: Is it possible to before step 2, to write my own script after step 1 and before step 2 to grep the sync'd emails on the disk and get rid of the ones I no longer want? Do you have any type of example about how to do the deleting part?

**Comment by gilleslamiral** [comment]
No local step with imapsync.

Filtering is possible although, read:
https://imapsync.lamiral.info/FAQ.d/FAQ.Messages_Selection.txt

https://imapsync.lamiral.info/FAQ.d/FAQ.Archiving.txt
````
=================================================================
Q. Can imapsync be used to maintain a local offline copy of a 
   mailbox from host1, eg for backup purposes, so that if the 
   server fails, then the mailbox could be reinstated,
   ideally without requiring a second IMAP host?

R1. No. Imapsync plays with imap servers only.

R2. Imapsync can't but rick-sanders-imap-tools can!
    See https://github.com/andrewnimmo/rick-sanders-imap-tools
    
    Back up and restore IMAP accounts with imapdump.pl and dumptoIMAP.pl
    It looks like imapdump.pl has options to avoid duplicates and be
    in an update mode. There are options -u -U
    [-u] Don't dump messages already dumped
    [-U] Don't dump message if it already exists in the dump directory
    I wonder the difference, I've never used Rick's tools.

    Remember that doing backups without trying the restore process
    is more dangerous than doing no backups at all. No backups makes
    people act in a safer way, well, usually...
    
R3. Use ImportExportTools for Mozilla Thunderbird by Paolo Kaosmos. 
    ImportExportTools does not do IMAP as a destination.

==================================================================
````

**Comment by simonhf** [comment]
Thanks for the answer and pointing me towards Rick Sanders IMAP tools, which seems to have the functionality I'm after.

I noticed that Rick Sanders is sadly no longer with us. Do you know of a user group for his tools? Or do you know whether they still work?

Also, would you ever consider consider adding the backup and restore functionality to your tool?

**Comment by gilleslamiral** [comment]
Rick Sanders IMAP tools work well as far as I know. 
The Rick mailing-lists are down. Use github to talk to users.

Adding backup/restore from disk is not on my roadmap.
